tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -204,6 +205,90 @@ private static void AgentConvertSpan<TInner>(TInner activity, Span span)
             {
                 span.Type = activity5 is null ? SpanTypes.Custom : AgentSpanKind2Type(activity5.Kind, span);
             }
+
+            // extract any ActivityLinks
+            ExtractActivityLinks<TInner>(span, activity5);
+        }
+
+        private static void ExtractActivityLinks<TInner>(Span span, IActivity5? activity5)
+            where TInner : IActivity
+        {
+            if (activity5 is null)
+            {
+                return;
+            }
+
+            foreach (var link in (activity5.Links))
+            {
+                var duckLink = link.DuckCast<IActivityLink>();
+
+                _ = HexString.TryParseTraceId(duckLink!.Context.TraceId.TraceId!, out var newActivityTraceId);
+                _ = HexString.TryParseUInt64(duckLink.Context.SpanId.SpanId!, out var newActivitySpanId);
+                var traceParentSample = duckLink.Context.TraceFlags > 0;
+                var traceState = W3CTraceContextPropagator.ParseTraceState(duckLink.Context.TraceState!);
+
+                var samplingPriority = traceParentSample switch
+                {
+                    true when traceState.SamplingPriority is > 0 => traceState.SamplingPriority.Value,
+                    true => SamplingPriorityValues.AutoKeep,
+                    false when traceState.SamplingPriority is <= 0 => traceState.SamplingPriority.Value,
+                    false => SamplingPriorityValues.AutoReject,
+                };
+
+                var spanContext = new SpanContext(
+                    newActivityTraceId,
+                    newActivitySpanId,
+                    samplingPriority: samplingPriority,
+                    serviceName: null,
+                    origin: traceState.Origin,
+                    isRemote: duckLink.Context.IsRemote);
+
+                var traceTags = TagPropagation.ParseHeader(traceState.PropagatedTags);
+
+                if (traceParentSample && traceState.SamplingPriority <= 0)
+                {
+                    traceTags.SetTag(Tags.Propagated.DecisionMaker, "-0");
+                }
+                else if (!traceParentSample && traceState.SamplingPriority > 0)
+                {
+                    traceTags.RemoveTag(Tags.Propagated.DecisionMaker);
+                }
+
+                spanContext.AdditionalW3CTraceState = traceState.AdditionalValues;
+                spanContext.LastParentId = traceState.LastParent;
+                spanContext.PropagatedTags = traceTags;
+
+                var duckLinkSpan = new Span(spanContext, DateTimeOffset.Now, new CommonTags());
+                var eventJObject = new JObject();
+                var spanLink = span.AddSpanLink(duckLinkSpan);
+
+                if (duckLink.Tags is not null)
+                {
+                    foreach (var kvp in duckLink.Tags)
+                    {
+                        if (!string.IsNullOrEmpty(kvp.Key)
+                         && IsAllowedAtributeType(kvp.Value!))
```

Comment:
The `!` asserts that `kvp.Value` cannot be null, but I don't think that's an assumption we should make. I think it would be better if we pass this in simply as `kvp.Value`, update the `IsAllowedAtributeType` method to a accept a `object?` parameter, and let it handle null values.

```suggestion
                         && IsAllowedAtributeType(kvp.Value))
```